velox/common/caching/tests/CachedFactoryTest.cpp

```diff
@@ -225,3 +225,33 @@ TEST(CachedFactoryTest, retrievedCached) {
   }
   EXPECT_EQ(*generated, 5);
 }
+
+TEST(CachedFactoryTest, disableCache) {
+  auto generator = std::make_unique<DoublerGenerator>();
+  auto* generated = &generator->generated_;
+  CachedFactory<int, int, DoublerGenerator> factory(
+      nullptr, std::move(generator));
+
+  auto val1 = factory.generate(1);
+  EXPECT_EQ(val1, cacheMiss(2));
+  EXPECT_EQ(*generated, 1);
+
+  auto val2 = factory.generate(1);
+  EXPECT_EQ(val2, cacheMiss(2));
+  EXPECT_EQ(*generated, 2);
+
+  EXPECT_EQ(factory.currentSize(), 0);
+
+  EXPECT_EQ(factory.maxSize(), 0);
+
+  EXPECT_EQ(factory.cacheStats(), SimpleLRUCacheStats(0, 0, 0, 0));
+
+  EXPECT_EQ(factory.clearCache(), SimpleLRUCacheStats(0, 0, 0, 0));
+
+  std::vector<int> keys;
+  std::vector<std::pair<int, int>> cached;
+  std::vector<int> missing;
+  factory.retrieveCached(keys, &cached, &missing);
```

Comment:
Why are you passing empty keys here? This doesn't seem to test cache-disabled code path since the results will be same regardless of whether cache is enabled or not.